[jmxfetch] Bump jmxfetch version from 0.42.0 to 0.42.1

sgnn7 opened the pull request:
### What does this PR do?

Bumps version of `jmxfetch` to 0.42.1. This version contains no user-visible changes in functionality except a different source now for the underlying artifact (sonatype vs bintray).

### Motivation

We use a different pipeline for publishing artifacts now so this change uses the first artifact produced that way

### Additional Notes

Anything else we should know when reviewing?

### Describe your test plan

- Run jmxfetch check
- Ensure that it works

sgnn7 committed:
[jmxfetch] Bump JMXFetch to 0.42.1

This release is tied to a different publishing pipeline but no
user-facing changes.

olivielpeau commented on releasenotes/notes/jmxfetch-0.42.1-a256f9b959d32952.yaml:
```diff
@@ -0,0 +1,12 @@
+---
+other:
+  - |
+    JMXFetch upgraded from `0.42.0` to `0.42.1` as a part of the move to
```
```suggestion
    JMXFetch upgraded from ``0.42.0`` to ``0.42.1`` as a part of the move to
```

sgnn7 committed:
[build] Use Sonatype for `jmxfetch` artifact download endpoint

Since our jmxfetch artifacts no longer get pushed to bintray, we are switching
to Sonatype as the source for this artifact.

olivielpeau reviewed (approved):
Thanks, LGTM!

Can you update the PR description to reflect that this makes the build pull from sonatype now?

sgnn7 commented:
Gitlab passed everything and the integration tests are failing due to a known-but-unrelated (`decorator ` in `datadogpy`) issue. Merging.
